met-web/src/components/engagement/form/EngagementFormTabs/EngagementTabsContext.tsx

```diff
@@ -240,13 +280,15 @@ export const EngagementTabsContextProvider = ({ children }: { children: React.Re
             ) as PatchEngagementSettingsRequest;
 
             if (Object.keys(updatedSettings).length === 0) {
+                setSaving(false);
                 return Promise.resolve();
```

Comment:
I don't think we need to call `Promise.resolve()` here. Within an `async/await` function, a promise will always be returned automatically.

```suggestion
                return;
```